src/v1/sources/shopify/webhookTransformations/serverSideUtlis.js

```diff
@@ -54,8 +58,73 @@ const getAnonymousIdFromAttributes = (event) => {
   return rudderAnonymousIdObject ? rudderAnonymousIdObject.value : null;
 };
 
+/**
+ * Returns the cart_token from the event message
+ * @param {Object} event
+ * @returns {String} cart_token
+ */
+const getCartToken = (event) => event?.cart_token || null;
+
+/**
+ * Handles the anonymousId assignment for the message, based on the event attributes and redis data
+ * @param {Object} message
+ * @param {Object} event
+ */
+const handleAnonymousId = async (message, event) => {
+  const anonymousId = getAnonymousIdFromAttributes(event);
+  if (isDefinedAndNotNull(anonymousId)) {
+    message.anonymousId = anonymousId;
+  } else {
+    // if anonymousId is not present in note_attributes or note_attributes is not present, query redis for anonymousId
+    const cartToken = getCartToken(event);
+    if (cartToken) {
```

Comment:
Lets add log in case `cartToken` is missing. We should log for which sourceId the `cartToken` was missing

Reply:
added a stat for the same instead